lib/errorlogger.cpp

```diff
@@ -1022,9 +1040,13 @@ std::string getGuideline(const std::string &errId, ReportType reportType,
                            guideline.begin(), static_cast<int (*)(int)>(std::toupper));
         }
         break;
-    case ReportType::misraC:
+    case ReportType::misraC2012:
+    case ReportType::misraC2023:
+    case ReportType::misraC2025:
         if (errId.rfind("misra-c20", 0) == 0 || errId.rfind("premium-misra-c-20", 0) == 0)
             guideline = errId.substr(errId.rfind('-') + 1);
+        if (errId.find("dir") != std::string::npos)
+            guideline = "Dir " + guideline;
```

Comment:
It seems to me if the errId is `abdirxy` then we will say that the guideline is "Dir " that seems bad.

I would prefer that it says `find("-dir-")` to make it more explicit.